### ISSUE TITLE
doc/ref: incorrect examples in bounds documentation

### ISSUE THREAD

**Issue by antong** [cue-lang/cue#1339]
Currently (v0.4.0), the language reference section for bounds (https://github.com/cue-lang/cue/blob/v0.4.0/doc/ref/spec.md#bounds), has the following table of examples:

```
2 & >=2 & <=5           // 2, where 2 is either an int or float.
2.5 & >=1 & <=5         // 2.5
2 & >=1.0 & <3.0        // 2.0
2 & >1 & <3.0           // 2.0
2.5 & int & >1 & <5     // _|_
2.5 & float & >1 & <5   // 2.5
int & 2 & >1.0 & <3.0   // _|_
2.5 & >=(int & 1) & <5  // _|_
>=0 & <=7 & >=3 & <=10  // >=3 & <=7
!=null & 1              // 1
>=5 & <=5               // 5
```

However, many of these are not what the current implementation (v0.4.0) gives. The following is a CUE representation of the example table, with the examples as fields and the original comments from the spec preserved. Where the actual result differs from the example in the spec, I added the real result to the comment.

```
t01: 2 & >=2 & <=5           // 2, where 2 is either an int or float. -- Actually the result is the int 2 and never a float.
t02: 2.5 & >=1 & <=5         // 2.5
t03: 2 & >=1.0 & <3.0        // 2.0 -- The actual result is 2
t04: 2 & >1 & <3.0           // 2.0 -- The actual result is 2
t05: 2.5 & int & >1 & <5     // _|_
t06: 2.5 & float & >1 & <5   // 2.5
t07: int & 2 & >1.0 & <3.0   // _|_ -- The actual result is 2
t08: 2.5 & >=(int & 1) & <5  // _|_ -- The actual result is 2.5
t09: >=0 & <=7 & >=3 & <=10  // >=3 & <=7
t10: !=null & 1              // 1
t11: >=5 & <=5               // 5
```

I think it is the examples that are wrong, and not the implementation.  Some of these examples seem redundant (t05--t08), and examples could be added for conflicting bounds and the case where the fact that the value is an integer makes the bound reduce to a concrete integer. I'm thinking the list could be changed to have the following examples (examples 5-8 removed, and 12 and 13 added):
```
t01: 2 & >=2 & <=5           // 2
t02: 2.5 & >=1 & <=5         // 2.5
t03: 2 & >=1.0 & <3.0        // 2
t04: 2 & >1 & <3.0           // 2
t09: >=0 & <=7 & >=3 & <=10  // >=3 & <=7
t10: !=null & 1              // 1
t11: >=5 & <=5               // 5
t12: >=2 & <2                // _|_
t13: int & >=1.7 & <3        // 2
```

I'm a bit puzzled by `t11: >=5 & <=5`. The original example gives 5, and so does v0.4.0 export to cue and cue def. But really it could be either the int 5 or the float 5.0. I think that is a separate implementation bug.

**Comment by myitcv** [comment]
Thanks for raising this @antong. I'm tempted to fold this into https://github.com/cue-lang/cue/issues/1340 which you kindly opened, because we should fundamentally fix the implementation and spec examples together.

**Comment by antong** [comment]
> I'm tempted to fold this into #1340 which you kindly opened, because we should fundamentally fix the implementation and spec examples together.

Agreed. I opened #1340 separately however, as this particular example (`>=5 & <= 5`) actually matches the output in the table (but clearly does have an implementation issue). The other examples I guess have just become wrong at some point. I think I also saw other examples in the spec where it seems it is assumed that literals like 5 can unify with float.

I read through the language reference during the weekend and fixed what I thought were obvious typos and errors (https://review.gerrithub.io/c/cue-lang/cue/+/526404). I opened this issue about the bounds section here for discussion as I wasn't really sure that I correctly understood what was wrong and it wasn't just clear typos.
